### PR TITLE
[EPLB] Expert histogram logging

### LINKED ISSUE #21038: [Feature]: Record EPLB metric, can in /metric api get info
### 🚀 The feature, motivation and pitch

In https://github.com/vllm-project/vllm/pull/18343#issuecomment-3011349148 this comment,  i think we can export some metrics, such as the weight of each expert, to understand the hot and cold distribution of experts, etc.

### Alternatives

_No response_

### Additional context

_No response_

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

### PR DESCRIPTION
**Pull request by PatrykSaffer**
## Purpose

Track expert metrics in MoEs.
This tracks experts usage per layer, rank usage per layer.
This adds ~1-2% overhead e2e.
Example results in grafana:
balancedness

## Test Plan
added test test_expert_usage_histogram.py

## Test Result
passed

### CODE REVIEW

File: vllm/v1/worker/gpu_model_runner.py [hunk 1]
```diff
@@ -2016,6 +2032,80 @@ def pad_out_ubatch_slice(ubatch_slices: UBatchSlices, num_total_tokens: int):
...
+                    compute_hash()]
+            else:
+                eplb_state = self.eplb_state.model_states[
+                    self.model_config.compute_hash()]
+
+            # When eplb enabled remap physical to logical experts.
+            logical_expert_usage_histogram = torch.zeros(
+                [
+                    self.model_config.get_total_num_moe_layers(),
+                    self.model_config.get_total_num_experts()
```
[the remark this answers is not in the document]
Comment (PatrykSaffer) [reply vllm/v1/worker/gpu_model_runner.py]: nice one, done

**Comment by abmfy** [comment]
This looks pretty cool! Please ping me when it’s ready for review.
